File: CreationClasses/src/Metier/Professeur.java
package Metier;

import java.util.Objects;

public class Professeur extends Personne {

    private double salaire;

    public Professeur(String nom, double salaire) {
        super(nom);
        this.salaire = salaire;

    }

    @Override
    public String afficher() {
        return getNom();
    }

    @Override
    public boolean equals(Object o) {
        Professeur other = (Professeur)o;
        return other.salaire == salaire && other.getNom() == getNom();
    }

    @Override
    public int hashCode() {
        return Objects.hash(getNom()) * Objects.hash(salaire);
    }
}

File: CreationClasses/metier/professeur.py
from metier.personne import Personne

class Professeur (Personne):

    def __init__(self, nom: str, salaire: int) -> None:
        super().__init__(nom)
        self.__salaire = salaire

    def afficher(self) -> str:
        return self.get_nom()

    def __eq__(self, o: object) -> bool:
        return self.__salaire == o.__salaire and self.get_nom() == o.get_nom()

    def __hash__(self) -> int:
        return self.get_nom().__hash__() * self.__salaire.__hash__()




